core/domain/state_domain.py

```diff
@@ -608,13 +616,13 @@ class InteractionInstance(translation_domain.BaseTranslatableObject):
     # of values, we used Any here.
     def __init__(
         self,
-        interaction_id: str,
+        interaction_id: Optional[str],
```

Comment:
Why is this `Optional` ?

Reply:
In the implementation of `to_dict` and `from_dict` method self.id is considered as Optional[str]:
https://github.com/oppia/oppia/blob/0e69c2bcd7a91ef3595da274d060608efd3140ae/core/domain/state_domain.py#L700
https://github.com/oppia/oppia/blob/0e69c2bcd7a91ef3595da274d060608efd3140ae/core/domain/state_domain.py#L743

Reply:
Why is it optional there?

Reply:
Yea, but I think the question still stands.

Reply:
While creating default interaction, interaction_id is always initialized as None:
https://github.com/oppia/oppia/blob/135e85b4419bd022b9da28718402697b28362262/core/domain/state_domain.py#L973